server/context.go

```diff
@@ -204,6 +211,18 @@ func (ctx *Context) parseGossipBootstrapResolvers() ([]gossip.Resolver, error) {
 	return bootstrapResolvers, nil
 }
 
+// GetHTTPClient returns the context http client, initializing it
+// if needed. It uses the context Certs.
+// TODO(marc): are there odd cases where may need a mutex?
```

Comment:
Once means also saving the error return which isn't much better.
This also shouldn't be a very contentious (or frequent) code path, so a mutex would be fine. I just don't think we need it for now.